src/auro-dropdown.js

```diff
@@ -94,7 +94,32 @@ class AuroDropdown extends LitElement {
     this.dropdownWidth = this.getBoundingClientRect().width;
   }
 
+  /**
+   * @private
+   * @returns {void} Updates the trigger `aria-label` or `aria-labeledby`.
+   */
+  processAriaLabel() {
+    this.trigger = this.shadowRoot.querySelector(`#trigger`);
+
+    if (this.getAttribute('aria-label')) {
+      // If custom aria-label declared at root apply it to the trigger
+      this.trigger.setAttribute('aria-label', this.getAttribute('aria-label'));
```

Comment:
I don't like imperatively setting the attribute like this. Can we do it in the template instead?

Reply:
Can you expand on a reason to move this more than "don't like"?

We can certainly move it to the template but I think that code is likely heavier and requires a little more mental leaps to account for the logic of when to use `label` vs `labelledby` and not end up with both attributes on the trigger.

Reply:
Sorry, I can clarify - I'm fine with keeping the if/else logic here. But I don't think we should be imperatively setting attributes inside the component with `this.trigger.setAttribute`. In general, if we're modifying the component's template, we should do it in the render method. Doing it imperatively in a different method can make it difficult to track down where it's happening. Instead, we should set a state variable and then apply it in the render method. So I'd prefer something like this... 

```
else if (this.getAttribute('aria-labelledby')) {
      // Else if custom aria-labelledby declared at root apply it to trigger
      this.labelledBy = this.getAttribute('aria-labelledby');
    }
// ...then in the render method

<div id="trigger" class="trigger" data-trigger-placement="${this.placement}" aria-labelledby=${this.labelledBy}>
```

I think doing it this way is also more performant, since Lit can optimize the state updates, though I'm not 100% certain.

Initializing `this.labelledBy` to undefined should make sure the attribute is not set by default (IIRC).

Reply:
@geoffrich This change was achieved using `ifDefined` rules in the template along with the change to reference properties rather than apply the attributes directly with JavaScript.